app/models/session.rb

```diff
@@ -4,14 +4,31 @@ class Session < ApplicationRecord
   validates :user_agent, length: { in: 0..255, allow_nil: true }
   validates :api_version, inclusion: { in: %w(20200115) }
 
-  has_secure_token :access_token
-  has_secure_token :refresh_token
+  attr_accessor :access_token
+  attr_accessor :refresh_token
 
-  before_create :extend_expiration_dates
+  before_create do
+    generate_tokens
+    extend_expiration_dates
+  end
 
   ACCESS_TOKEN_AGE = Rails.application.config.x.session[:access_token_age].seconds
   REFRESH_TOKEN_AGE = Rails.application.config.x.session[:refresh_token_age].seconds
 
+  def self.authenticate(request_token)
+    _version, session_id, access_token = request_token.split(':')
+    session = Session.find_by_uuid(session_id)
+    if session && !access_token.nil?
+      hashed_access_token = Digest::SHA256.hexdigest(access_token)
```

Comment:
Seems like this should use `create_hash_from_value`

Reply:
`create_hash_from_value` is an instance method while `self.authenticate` is static.